### DISCUSSION TITLE
Fermentables and the mashing/mashability thereof

### DISCUSSION THREAD

**Discussion by matty0ung** [Brewtarget/brewtarget#700] [Q&A] [answered]
I have a couple of quick questions about Fermentable (which stem from adding support for BeerJSON):

- There seem to be two similar properties `isMashed` and `recommendMash`.  What's the intended difference between them?  Does the former apply only to the specific use of a fermentable in a particular recipe?
- On a related note, we have `additionMethod` which seems to overlap with `isMashed` but extend it to say a not-mashed grain is "steeped" while a not-mashed non-grain is simply "not mashed".  Have I understood this right?

**Comment by mikfire** [comment, marked as answer]
As I recall, the isMashed is used to modify the sugar calculations and to change the generated instructions and is expected to be used as part of a recipe -- that is, the cloned fermentable. we also assume that a thing that is a grain and not mashed must be steeped. I am not sure I agree with that assumption.

There were a number of attributes we maintained only because they were part of the XML "specification". I suspect recommendMash and additionMethod are two of them, because searching the code doesn't show either being used other than to toggle the switches in the editor. This isn't to say people aren't using them, it is simply that they have no impact outside of the editors.

A quick dig into the database suggests this is also true. I am showing things like Crystal 10 as having recommend_mash set to false, and Briess Crystal 10 as having both is_mashed and recommend_mash set to false.

**Reply by matty0ung** [reply]
Thanks Mik!
